src/iosched/unified.c

```diff
@@ -1001,7 +1075,9 @@ int unified_truncate(struct dentry *d, off_t length, void *iosched_handle)
 	ltfs_mutex_unlock(&d->iosched_lock);
 	releaseread_mrsw(&priv->lock);
 
-	if (! dpr)
+	if (dpr)
+		_unified_put_dentry_priv(dpr, priv);
+	else
```

Comment:
I think it's worth adding a comment about why we only truncate when `dpr == NULL`

Reply:
Hmm, it is little bit nasty.  I just added the line with respecting the original code.

It looks this `ltfs_fsraw_truncate` call can be merged to the previous block like 

```
	_unified_get_dentry_priv(d, &dpr, priv);
	if (dpr) {
		...
		/* Tell libltfs about the truncate request */
		ltfs_mutex_lock(&dpr->io_lock);
		ret = ltfs_fsraw_truncate(d, length, priv->vol);
		ltfs_mutex_unlock(&dpr->io_lock);

		_unified_put_dentry_priv(dpr, priv);
	} else {
		/* No corresponded DPR just call normal truncate */
		ret = ltfs_fsraw_truncate(d, length, priv->vol);
	}

	ltfs_mutex_unlock(&d->iosched_lock);
	releaseread_mrsw(&priv->lock);
```

I don't know why call `ltfs_fsraw_truncate` sfter unlocking d->iosched_lock and priv->lock. It looks we want to make the time to have locks as short as we can. But I think this logic much easy to understand.